discover/action.yml

```diff
@@ -20,17 +20,21 @@ runs:
         delim=$RANDOM
 
         cat << $delim >> "$GITHUB_ENV"
-        DISC_PATH=${{ runner.temp }}/discovery
-        DISC_ARC_PATH=${{ runner.temp }}/discovery.tar.zstd
         $delim
       shell: bash
 
     - name: Cache Nix Store
       id: cache-nix
-      uses: divnix/nix-cache-action@v3.0.11-nix
+      uses: actions/cache@v3
       with:
-        path: /nix
-        key: ${{ runner.os }}-${{ github.repository }}-discovery
+        path: |
+          /nix
+          ${{ env.DISC_ARC_PATH }}
+        key: discovery-${{ runner.os }}-${{ github.ref }}-${{ github.sha }}
+        restore-keys: |
+          discovery-${{ runner.os }}-${{ github.ref }}
```

Comment:
would (hopefully) prefer the same branch :pray: -- but that isn't eminent from the docs